### PR TITLE
refactor(test): make CORS test consistent with expected behavior

### LINKED ISSUE #4881: Maintenance: cors is opt out if enabled
### Why is this needed?

Some cors tests did not opt out for non-cors endpoints.

```python
def test_cors_allow_all_request_origins():
    # GIVEN a function with cors=True
    # AND http method set to GET
    app = ApiGatewayResolver(
        cors=CORSConfig(
            allow_origin="*",
            allow_credentials=True,
        ),
    )

    @app.get("/my/path", cors=True)
    def with_cors() -> Response:
        return Response(200, content_types.TEXT_HTML, "test")

    @app.get("/without-cors")
    def without_cors() -> Response:
        return Response(200, content_types.TEXT_HTML, "test")
```

### Which area does this relate to?

Tests

### Solution

opt out cors instead of opt-in.

```python
def test_cors_allow_all_request_origins():
    # GIVEN a function
    # AND http method set to GET
    app = ApiGatewayResolver(
        cors=CORSConfig(
            allow_origin="*",
            allow_credentials=True,
        ),
    )

    @app.get("/my/path")
    def with_cors() -> Response:
        return Response(200, content_types.TEXT_HTML, "test")

    @app.get("/without-cors", cors=False)
    def without_cors() -> Response:
        return Response(200, content_types.TEXT_HTML, "test")
```

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by tonnico**
**Issue number: #4881**

### Changes

opt-out cors instead of opt-in.

### User experience

when a resolver has cors enabled, routes can opt out only

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: tests/functional/event_handler/required_dependencies/test_api_gateway.py [hunk 1]
```diff
@@ -331,7 +331,7 @@ def test_cors():
     def with_cors() -> Response:
```
Comment (leandrodamascena) [inline tests/functional/event_handler/required_dependencies/test_api_gateway.py:331]: I'm wondering if we can remove `cors=True` here. It's the default behavior when cors is set in `ApiGatewayResolver`. What do you think?

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Thanks for sending this PR @Wurstnase, great catch!!! I just left a quick comment before we merge.

Thanks

File: tests/functional/event_handler/required_dependencies/test_api_gateway.py (hunk 1 above)
Comment (tonnico) [reply tests/functional/event_handler/required_dependencies/test_api_gateway.py:331]: Good point. I will check and test some other places to.

File: tests/functional/event_handler/required_dependencies/test_api_gateway.py
Comment (tonnico) [inline tests/functional/event_handler/required_dependencies/test_api_gateway.py:1]: low hanging fruit to test the UserWarnings and reduce noise in the pytest output

Comment (leandrodamascena) [reply tests/functional/event_handler/required_dependencies/test_api_gateway.py:1]: Great attention to details @Wurstnase! Thank you so much!! ❤️

**Review by leandrodamascena — APPROVED** [review]
APPROVED! 🚀 

Thanks @Wurstnase

**Pull request merged**